class-gf-razorpay.php

```diff
@@ -0,0 +1,319 @@
+            'name'        => $form['name'],
+            'amount'      => (int) round($entry['payment_amount'] * 100),
+            'currency'    => $entry['currency'],
+            'description' => $form['description'],
+            'prefill'     => array(
+                'name'    => $customerFields['billingInformation_firstName'],
+                'email'   => $customerFields['billingInformation_email'],
+            ),
+            'notes'       => array(
+                'gravity_forms_order_id' => $entry['id']
+            ),
+            'order_id'    => $entry['razorpay_order_id'],
+        );
+
+        $redirectUrl = '?page=gf_razorpay_callback';
+
+
+        wp_enqueue_script('razorpay_script',
+                          plugin_dir_url(__FILE__). 'script.js',
+                          array('checkout')
+        );
+
+        wp_localize_script('razorpay_script',
+                           'razorpay_script_vars',
+                           array(
+                               'data' => $razorpayArgs
+                           )
+        );
+
+        wp_register_script('checkout',
+                           'https://checkout.razorpay.com/v1/checkout.js',
+                           null,
+                           null);
+
+        wp_enqueue_script('checkout');
+
+        return $this->generate_order_form($redirectUrl);
+    }
+
+    function generate_order_form($redirectUrl)
+    {
+        $html = <<<EOT
+<form id ='razorpayform' name='razorpayform' action="$redirectUrl" method='POST'>
+    <input type='hidden' name='razorpay_payment_id' id='razorpay_payment_id'>
+    <input type='hidden' name='razorpay_signature'  id='razorpay_signature' >
+</form>
+<p id='msg-razorpay-success'  style='display:none'>
+    Please wait while we are processing your payment.
+</p>
+<p>
+    <button id='btn-razorpay'>Pay Now</button>
+    <button id='btn-razorpay-cancel' onclick='document.razorpayform.submit()'>Cancel</button>
+</p>
+EOT;
+        return $html;
+    }
+
+    public function is_callback_valid()
+    {
+        //Will check if the return url is valid
+        if (rgget('page')!== 'gf_razorpay_callback')
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function pay_using_razorpay($entry, $form)
+    {
+        $feed = $this->get_payment_feed($entry, $form);
+
+        $customerFields = $this->get_customer_fields_array($feed, $entry);
+
+        //gravity form method to get value of payment_amount key from entry
+        $paymentAmount = rgar($entry, 'payment_amount' );
+
+        //It will be null first time in the entry
+        if (empty($paymentAmount) === true)
+        {
+            $paymentAmount = GFCommon::get_order_total($form, $entry);
+            gform_update_meta($entry['id'], 'payment_amount', $paymentAmount );
+            $entry['payment_amount'] = $paymentAmount;
+        }
+
+        $api = new Api($this->get_plugin_setting('gf_razorpay_key'),
```

Comment:
Please move key and secret to different variables. Makes the line too long. Would be nice to have a simple Api declaration.